addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/MainTVServerService.java

```diff
@@ -176,7 +176,7 @@ public void onServiceSubscribed(String service, boolean succeeded) {
     public void onValueReceived(String variable, String value, String service) {
 
         String oldValue = stateMap.get(variable);
-        if (value.equals(oldValue)) {
+        if (oldValue.equals(value)) {
```

Comment:
Yes it would "fall apart" but the intention was to ignore it (i guess)...
Your solution looks good to me. :+1: